terraform/ingest/index.js

```diff
@@ -0,0 +1,261 @@
+          ethnicityId,
+          percent: ethnicityTotal / (salaryBucketTotals[dateString][bucket] || 1)
+        }
+      })
+      summaries.push(...ethnicitySummaries)
+    })
+  }
+
+  const putObjectParams = [
+    [summaries, 'summaries.json'],
+    [DEPARTMENT_IDS_TO_NAMES, 'departments.json'],
+    [Object.keys(employeesByDate), 'dates.json']
+  ]
+
+  const putObjectPromises = putObjectParams.map(([data, filename]) => {
+    var params = {
+      Body: JSON.stringify(data),
+      Bucket: S3_BUCKET,
+      Key: `data/${filename}`,
+      ACL: `public-read`
+    }
+    s3.putObject(params).promise()
+  })
+
+  const employeesByDatePromises = Object.keys(employeesByDate).map(dateString => {
+    var params = {
+      Body: csvFormat(employeesByDate[dateString]),
+      Bucket: S3_BUCKET,
+      Key: `data/${dateString}/employees.csv`,
+      ACL: `public-read`
+    }
+    s3.putObject(params).promise()
+  })
+  return Promise.all([...putObjectPromises, ...employeesByDatePromises])
+}
+
+function employeeFromCSVLine (employee) {
+  const salary = parseSalary(employee['Annual Salary'])
+  const salaryBucketId = bucketSalary(salary)
+
+  return {
+    ethnicityId: parseInt(employee['Ethnic Code']),
+    salaryBucketId,
+    departmentId: getDepartmentId(employee['Current Dept Description'] || employee['Department']),
+    gender: employee['Gender']
+  }
+}
+
+/*
+    Generate a dynamic list of codes to departments, returnin the ID of the
+    passed in department.
+
+    We can save this and use it for efficient lookups on the client side.
+*/
+function getDepartmentId (department) {
+  if (!DEPARTMENT_NAMES_TO_IDS[department]) {
+    DEPARTMENT_NAMES_TO_IDS[department] = departmentId
+    DEPARTMENT_IDS_TO_NAMES[departmentId] = department
+    departmentId += 1
+  }
+
+  return DEPARTMENT_NAMES_TO_IDS[department]
+}
+
+function bucketSalary (salary) {
+  if (salary < 33000) return constants.LOW
+  if (salary < 66000) return constants.MID
+  return constants.HIGH
+}
+
+/*
+Internal method to parse the income string we get back from reports into a float
+
+Some seasonal employees have "Annual Salary" listed as an empty string, and
+some records in 20150301.csv just have '$-  ' listed as a salary. Allow these
+to raise an exception.
+
+salary - String - a dollar amount with thousands separators and dollar sign
+    e.g. '$1,324.00'. May also be just '$-'
+*/
+function parseSalary (salary) {
+  try {
+    const cleaned = salary.replace(',', '').replace('$', '')
+    return parseFloat(cleaned)
+  } catch (ex) {
+    console.warn(`Exception ${ex}`)
+    return 0
+  }
+}
+
+/**
+ * HTTP requests for latest published General Government Employees Demographics dataset.
+ */
+function fetchPublishedData () {
+  var filename
+  return fetch(DATA_URL)
+    .then(resp => resp.json())
+    .then(data => {
+      const rowsUpdatedAt = new Date(data['rowsUpdatedAt'] * 1000)
+      filename = rowsUpdatedAt.toISOString().replace(/-/g, '').slice(0, 8)
```

Comment:
nit: I'm just thinking we probably don't need to drop hyphens for a filename - it just makes it more to parse for someone else.

Reply:
Nice catch!